src/timing.h

```diff
@@ -313,12 +327,14 @@ STATIC_INLINE void _jl_timing_block_ctor(jl_timing_block_t *block, int owner, in
 }
 
 STATIC_INLINE void _jl_timing_block_destroy(jl_timing_block_t *block) JL_NOTSAFEPOINT {
+    _TRACY_DESTROY(block->tracy_ctx);
+    if (!block->enabled) return;
     uint64_t t = cycleclock(); (void)t;
-
-    _ITTAPI_STOP(block);
+    if (jl_timing_enabled(block->owner)) {
```

Comment:
Does this behave correctly if an owner is disabled in the middle of a timing block?

Reply:
I wouldn't recommend it, ITTAPI commits part of the change as soon as the constructor is called.

Reply:
Can we guarantee we do `_ITTAPI_STOP` if we did `_ITTAPI_START`?

I think the main confusion is that we have two enable conditions: one ignoring jl_timing_enabled which we use only for COUNTS, and another that considers jl_timing_enabled which we use for all of the proper profilers

Reply:
If we need to guarantee that, then I think it's by and large equivalent to saying we shouldn't disable after starting the event. I think any use case where that would be necessary can also be covered by multiple events, one unconditional at the start and a second conditional event when the condition becomes available.

Reply:
If we had an 'active' bit in the timing block, then we could safely disable the owner globally without skipping the destructor for the active event, couldn't we?